dist/gmusic.js

```diff
@@ -195,6 +204,47 @@ proto.playback = {
     this.playback._sliderEl.dispatchEvent(evt);
   },
 
+  // Return song duration in milliseconds
+  getSongDuration: function () {
+    return this.playback._sliderEl.max;
+  },
+
+  getPlaybackState: function () {
+    // Play/Pause element states:
+    //   PLAYING: {__data__: {icon: 'av:pause-circle-filled'}, disabled: false}
+    //   PAUSED: {__data__: {icon: 'av:sj:pause-circle-fill'}, disabled: false}
+    //   STOPPED: {__data__: {icon: 'av:sj:play-circle-fill'}, disabled: true}
+    if (!this.playback._playPauseEl.disabled) {
+      if (this.playback._playPauseEl.__data__.icon === 'av:pause-circle-filled') {
+        return GMusic.Playback.PLAYING;
+      } else {
+        return GMusic.Playback.PAUSED;
+      }
+    } else {
+      return GMusic.Playback.STOPPED;
+    }
+  },
+
+  getSongInfo: function () {
+    var songInfo = {};
+    songInfo.title = this.doc.getElementById(SELECTORS.info.titleId).textContent;
+    songInfo.artist = this.doc.getElementById(SELECTORS.info.artistId).textContent;
+    songInfo.album = this.doc.querySelector(SELECTORS.info.albumSelector).textContent;
+    songInfo.art = this.doc.getElementById(SELECTORS.info.albumArtId);
+    songInfo.duration = this.doc.getElementById(SELECTORS.playback.sliderId).max;
+
+    songInfo.title = (songInfo.title) ? songInfo.title : 'Unknown';
```

Comment:
Mostly personal preference here but I see these as better JS if it goes like

```js
songInfo.title = songInfo.title || 'Unknown'
```

In fact you could take it a step further and do

```js
songInfo.title = this.doc.getElementById(SELECTORS.info.titleId).textContent || 'Unknown';
```